frontend/app/services/subscription-service.server.ts

```diff
@@ -242,41 +277,111 @@ function createSubscriptionService() {
     }
   }
 
-  //TODO: this one will be updated in another PR
-  async function updateSubscription(sin: string, subscription: SubscriptionInfo) {
+  async function createSubscription(userId: string, subscription: SubscriptionInfo) {
     const auditService = getAuditService();
     const instrumentationService = getInstrumentationService();
-    auditService.audit('alert-subscription.update', { sin });
+    auditService.audit('alert-subscription.update', { userId });
+    const url = new URL(`${CDCP_API_BASE_URI}/api/v1/users/${userId}`);
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
 
-    // TODO: "IT-Security won't like SIN being passed as identifier"
-    // TODO: add CDCP_API_BASE_URI
-    const userId = sin;
-    const url = new URL(`https://api.cdcp.example.com/v1/users/${userId}/subscriptions`);
+    instrumentationService.countHttpStatus('http.client.cdcp-api.users.gets', response.status);
 
-    const response = await fetch(url, {
-      method: 'PUT',
+    if (!response.ok) {
+      log.error('%j', {
+        message: 'Failed find data',
+        status: response.status,
+        statusText: response.statusText,
+        url: url,
+        responseBody: await response.text(),
+      });
+
+      throw new Error(`Failed to find data. Status: ${response.status}, Status Text: ${response.statusText}`);
+    }
+
+    const user = await response.json();
+    const userParsed = userSchema.parse(user);
+
+    const userSubscriptionsURL = userParsed._links.subscriptions.href;
+    const subscriptionsResponse = await fetch(userSubscriptionsURL, {
+      method: 'POST',
       headers: {
         'Content-Type': 'application/json',
       },
       body: JSON.stringify(subscription),
     });
 
-    instrumentationService.countHttpStatus('http.client.cdcp-api.subscriptions.posts', response.status);
+    instrumentationService.countHttpStatus('http.client.cdcp-api.subscriptions.create', subscriptionsResponse.status);
 
-    if (response.status === 204) {
-      return null;
+    if (!subscriptionsResponse.ok) {
+      log.error('%j', {
+        message: 'Failed to create data',
+        status: subscriptionsResponse.status,
+        statusText: subscriptionsResponse.statusText,
+        url: url,
+        responseBody: await subscriptionsResponse.text(),
+      });
+
+      throw new Error(`Failed to create data. Status: ${subscriptionsResponse.status}, Status Text: ${subscriptionsResponse.statusText}`);
     }
+  }
+
+  async function updateSubscription(userId: string, subscription: SubscriptionInfo) {
+    const auditService = getAuditService();
+    const instrumentationService = getInstrumentationService();
+    auditService.audit('alert-subscription.update', { userId });
+
+    const url = new URL(`https://api.cdcp.example.com/api/v1/users/${userId}`);
+
+    const response = await fetch(url, {
```

Comment:
I think this should use the new `instrumentedFetch()` function: <https://github.com/DTS-STN/canadian-dental-care-plan/blob/main/frontend/app/utils/fetch-utils.server.ts#L55>

Reply:
Again, I'm not sure we need to fetch the user here. The user can be fetched at login and stored in session.